src/Model/Product/ProductData.php

```diff
@@ -11,38 +11,48 @@
 
 class ProductData implements JsonSerializable
 {
+    public const FEED_CONDITION_TYPE = 'ConditionType';
+    public const FEED_PACKAGE_WEIGHT = 'PackageWeight';
+    public const FEED_PACKAGE_HEIGHT = 'PackageHeight';
+    public const FEED_PACKAGE_WIDTH = 'PackageWidth';
+    public const FEED_PACKAGE_LENGTH = 'PackageLength';
+
     /**
-     * @var array
+     * @var mixed[]
      */
     protected $attributes = [];
 
-    public function __construct(string $conditionType, float $packageHeight, float $packageWidth, float $packageLength, float $packageWeight)
-    {
-        if (!in_array($conditionType, ProductConditionTypes::CONDITION_TYPES)) {
-            throw new InvalidDomainException('ConditionType');
+    public function __construct(
+        string $conditionType = null,
+        float $packageHeight = null,
+        float $packageWidth = null,
+        float $packageLength = null,
+        float $packageWeight = null
```

Comment:
II think you missed type as nullable.
```
        ?string $conditionType = null,
        ?float $packageHeight = null,
        ?float $packageWidth = null,
        ?float $packageLength = null,
        ?float $packageWeight = null
```

Reply:
It was intentional but it is OK to add it I suppose